test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java

```diff
@@ -54,15 +54,20 @@ public void testFromXContent() throws IOException {
     }
 
     /**
-     * Generic test that checks that if the test instance extends
-     * {@link ToXContentToBytes} the Strings.toString method renders the
+     * Generic test that checks the Strings.toString and/or toString() method
+     * (if the instance implements {@link ToXContentToBytes}) renders the
      * XContent correctly.
      */
     public void testToString() throws IOException {
         for (int runs = 0; runs < NUMBER_OF_TEST_RUNS; runs++) {
             T testInstance = createTestInstance();
             XContentType xContentType = XContentType.JSON;
-            String toString = Strings.toString(testInstance);
+            String toString;
+            if (testInstance instanceof ToXContentToBytes) {
+                toString = randomBoolean() ? Strings.toString(testInstance) : testInstance.toString();
+            } else {
+                toString = Strings.toString(testInstance);
+            }
```

Comment:
Given that we may want to drop ToXContentToBytes, I would drop this if too, just leave the else part as only few Writeable objects actually implement toString.

I would rename this test to testValidOutput and add a comment that explains why we have such a test. Now that I think about it again, this doesn't need to go through Strings.toString. WE can use XContentHelper.toXContent directly.